File: src/apps/ShramSafal/ShramSafal.Application/UseCases/Logs/AddLogTask/AddLogTaskCommand.cs
using ShramSafal.Domain.Logs;

namespace ShramSafal.Application.UseCases.Logs.AddLogTask;

public sealed record AddLogTaskCommand(
    Guid DailyLogId,
    string ActivityType,
    string? Notes,
    DateTime? OccurredAtUtc = null,
    Guid? LogTaskId = null,
    Guid ActorUserId = default,
    string? ActorRole = null,
    string? ClientCommandId = null,
    ExecutionStatus ExecutionStatus = ExecutionStatus.Completed,
    string? DeviationReasonCode = null,
    string? DeviationNote = null,
    // DATA_PRINCIPLE_SPINE sub-phase 04.3b — forensic provenance for the
    // emitted AuditEvent row. Sourced from HttpContext.AuditClaims() at the
    // endpoint; sentinel defaults keep direct-construction tests green.
    string ClientAppVersion = "unknown",
    string AuditDeviceId = "unknown",
    string AuditIpHash = "sha256:unknown");

File: src/clients/mobile-web/src/application/usecases/sync/AddLogTaskCommand.ts
import { mutationQueue } from '../../../infrastructure/sync/MutationQueue';
import { SyncMutationName } from '../../../infrastructure/sync/SyncMutationCatalog';

export interface AddLogTaskPayload {
     dailyLogId: string;
     logTaskId: string;
     activityType: string;
     notes?: string;
     occurredAtUtc?: string;
}

export class AddLogTaskCommand {
     static async enqueue(payload: AddLogTaskPayload): Promise<string> {
          const clientRequestId = `${SyncMutationName.AddLogTask}:${payload.dailyLogId}:${payload.logTaskId}`;
          return mutationQueue.enqueue(SyncMutationName.AddLogTask, payload, {
               clientRequestId,
               clientCommandId: clientRequestId
          });
     }
}
